### DISCUSSION TITLE
Use the same variable in the file name and inside the file itself

### DISCUSSION THREAD

**Discussion by deseven** [chhoumann/quickadd#573] [Q&A] [answered]
Hi, I'm trying to use the **Template** functionality of QuickAdd.

I created my template and a new template action in QuickAdd with file name format `{{DATE}} {{VALUE:myvalue}}`. It works as intended, QuickAdd asks me for `myvalue` and correctly puts it in the file name. However, I also need to use this variable inside the file. I tried adding the following inside my template:

```js
const myvalue = await this.quickAddApi.inputPrompt("myvalue");
return `${myvalue}`;
```

But this makes it ask for `myvalue` twice. And if I remove `inputPrompt()` it errors out because it has nothing to return.

Any way to make it work properly?

**Comment by chhoumann** [comment, marked as answer]
Sorry for the very late reply.

No script needed - QuickAdd's format syntax works inside the template file itself, and a named variable is asked for once per run and then reused everywhere it appears. Since your file name format already uses `{{VALUE:myvalue}}`, just put `{{VALUE:myvalue}}` in the template body and QuickAdd inserts the same value there without prompting again. `{{DATE}}` works in the body the same way.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax

**Reply by deseven** [reply]
Thank you, I personally no longer need it, but maybe someone would be happy to find the solution.
